### PR TITLE
Issue Fix - Quick Action:#6508

### PR DESCRIPTION
**Pull request by MadWasp79**
Issue Fix - Quick Action: Incorrect position when creating "My places" point #6508

 + Day/Night mode QA refactoring.

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java [hunk 1]
```diff
@@ -204,6 +206,7 @@ public boolean isWidgetVisible() {
      * @return true, if state was changed
      */
     public boolean setLayerState(boolean showWidget) {
+		prevWidgetState = showWidget;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java]: Space?

Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java]: formatted

File: OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java [hunk 2]
```diff
@@ -44,6 +44,7 @@
 import java.lang.ref.WeakReference;
 import java.util.ArrayList;
 import java.util.List;
+import net.osmand.plus.quickaction.actions.DayNightModeAction;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java]: That class should not be used in generic class MapQUickActionLayer

File: OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java [hunk 3]
```diff
@@ -73,6 +74,7 @@
     private boolean isLayerOn;
 
     private boolean nightMode;
+    private boolean prevWidgetState;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java]: Meaning is not clear, probably should be removed

Comment (vshcherb) [reply OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java]: This should stay but properly renamed, actually previous naming "showWidget" is also not clear and better naming should be used. It is not  "previous", I guess it is "current" state

File: OsmAnd/src/net/osmand/plus/quickaction/actions/DayNightModeAction.java [hunk 4]
```diff
@@ -25,14 +26,8 @@ public void execute(MapActivity activity) {
...
 			}
-			case NIGHT: {
-				activity.getMyApplication().getSettings().DAYNIGHT_MODE.set(OsmandSettings.DayNightMode.DAY);
-				break;
-			}
-			case AUTO: {
-				activity.getMyApplication().getSettings().DAYNIGHT_MODE.set(OsmandSettings.DayNightMode.DAY);
-				break;
-			}
+			case NIGHT:
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/DayNightModeAction.java]: Night, auto are switching to day which is strange. 
"DAY" (AUTO) -> "DAY" -> "NIGHT" [wrong]
"NIGHT" (AUTO) -> "DAY" -> "NIGHT" [correct]
"DAY" (AUTO) -> "NIGHT" -> "DAY" [correct]

Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/quickaction/actions/DayNightModeAction.java]: resolved

File: OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java (hunk 3 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/views/MapQuickActionLayer.java]: renamed it to currentWidgetState

**Pull request merged**